android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/SDLLockScreenActivity.java

```diff
@@ -158,19 +168,85 @@ public void initializeActivity(Intent intent){
 				String warningMsg = intent.getStringExtra(KEY_LOCKSCREEN_WARNING_MSG);
 				if (mIsDismissible) {
 					setLockscreenWarningMessage(warningMsg);
+				} else if (!useWhiteIconAndTextColor) {
+					setTextColorBlack();
 				}
 			}
 		}
 	}
 
-	private void changeBackgroundColor(int customColor) {
+	/**
+	 *  Sets the lockScreen logo
+	 */
+	private void setSdlLogo() {
+		ImageView lockScreen_iv = findViewById(R.id.lockscreen_image);
+		Drawable sdlIcon = getResources().getDrawable(R.drawable.sdl_lockscreen_icon);
+		// Checks color contrast and determines if the logo should be black or white
+		if (useWhiteIconAndTextColor) {
+			int color = Color.parseColor("#ffffff");
+
+			int red = (color & 0xFF0000) / 0xFFFF;
+			int green = (color & 0xFF00) / 0xFF;
+			int blue = color & 0xFF;
+
+			float[] matrix = {0, 0, 0, 0, red,
+					0, 0, 0, 0, green,
+					0, 0, 0, 0, blue,
+					0, 0, 0, 1, 0};
+
+			ColorFilter colorFilter = new ColorMatrixColorFilter(matrix);
+			sdlIcon.setColorFilter(colorFilter);
+		}
+		lockScreen_iv.setImageDrawable(sdlIcon);
+	}
+
+	/**
+	 * Changes the text color to white on the lockScreen
+	 */
+	private void setTextColorBlack() {
+		TextView tv = findViewById(R.id.lockscreen_text);
+		tv.setTextColor(Color.parseColor("#000000"));
+	}
+
+	/**
+	 *  Calculates the contrast of the background to determine if the Icon and Text color
+	 *  should be white or black
+	 * @return True if Background and Icon should be white, False if black
+	 */
+	private boolean shouldUseWhiteForegroundForBackgroundColor() {
+		float r = Color.red(backgroundColor) / 255f;
+		float b = Color.blue(backgroundColor) / 255f;
+		float g = Color.green(backgroundColor) / 255f;
+
+		// http://stackoverflow.com/a/3943023
```

Comment:
Do we typically reference stack overflow in the code like this?

Reply:
This code was pulled from IOS and minor changes were made to convert to Java. Since IOS has it, I left it